samcli/lib/sync/infra_sync_executor.py

```diff
@@ -272,6 +274,14 @@ def _auto_skip_infra_sync(
             LOG.debug("The current template is different from the last deployed version, we will not skip infra sync")
             return False
 
+        try:
+            if not self._param_overrides_subset_of_stack_params(stack_name, parameter_overrides):
+                LOG.debug("Detected changes between Parameter overrides and the current stack parameters.")
+                return False
+        except ClientError as ex:
+            LOG.debug("Unable to fetch stack Parameters from stack with name %s", stack_name, exc_info=ex)
+            return False
```

Comment:
This `ClientError` is part of `_param_overrides_subset_of_stack_params` function. No need to bubble it up here, please handle it inside the function and then return the appropriate response back to here.

Reply:
Fair, I'll move all error handling and logging into the parameter function